caffe2/core/nomnigraph/include/nomnigraph/Graph/Graph.h

```diff
@@ -80,23 +80,47 @@ class Node : public StorageType<T>, public Notifier<Node<T, U>> {
     inEdges.emplace_back(e);
   }
 
+  /// \brief Swaps an old in-edge for a new one.
+  /// \p oldE The old edge to be replaced.
+  /// \p newE A reference to an edge that will be added as an in-edge.
+  void replaceInEdge(EdgeRef oldE, EdgeRef newE) {
+    std::replace(inEdges.begin(), inEdges.end(), oldE, newE);
+    auto oldProducer = oldE->tail();
+    oldProducer->removeOutEdge(oldE);
```

Comment:
I see that the order of edges may change as a result, could it be a problem?

Reply:
this actually fixes the order of edges for the node being replaced but necessarily removes the old outEdge
```
a --> b
```
becomes
```
a -->
c --> b
```
and the dangling `-->` needs to be deleted.  if `a == c` in this case that is still fine